daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2126,9 +2157,11 @@ func (devices *DeviceSet) removeDevice(devname string) error {
 	return err
 }
 
-func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) error {
+func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) (bool, error) {
```

Comment:
There are few conditions where err is nil still deferred remove was not cancelled.

- If device was not scheduled for deferred removal.
- If device is not active to being with.

        if devinfo != nil && devinfo.DeferredRemove == 0 {
                return nil
        }